src/render.rs

```diff
@@ -53,15 +54,23 @@ pub struct Renderer<'a> {
     context: Value,
     tera: &'a Tera,
     for_loops: Vec<ForLoop>,
+    should_escape: bool,
 }
 
 impl<'a> Renderer<'a> {
     pub fn new(tpl: &'a Template, tera: &'a Tera, context: Value) -> Renderer<'a> {
+        let mut should_escape = false;
+        for ext in tera.autoescape_extensions.clone() {
```

Comment:
Why is there a `clone` here? Should be able to simply do:

`for ext in &tera.autoescape_extenions { ... }`